src/client/entity-editor/common/entity-linked-search-field.js

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (C) 2015-2017  Ben Ockmore
+ *               2015-2016  Sean Burke
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+// @flow
+
+import CustomInput from '../../input';
+import Entity from './entity';
+import LinkedEntity from './linked-entity';
+import React from 'react';
+import {Async as SelectAsync} from 'react-select';
+import ValidationLabel from '../common/validation-label';
+import _ from 'lodash';
+import makeImmutable from './make-immutable';
+import request from 'superagent-bluebird-promise';
+
+
+const ImmutableAsyncSelect = makeImmutable(SelectAsync);
+
+/**
+ * Determines whether an entity provided to the EntitySearch component is an
+ * Area, using the present attributes.
+ *
+ * @param {Object} entity the entity to test
+ * @returns {boolean} true if the entity looks like an Area
+ */
+function isArea(entity) {
+	if (entity.type === 'Area') {
+		return true;
+	}
+
+	if (entity.gid) {
+		return true;
+	}
+
+	return false;
+}
+
+type LanguageOption = {
+	name: string,
+	id: number
+};
+
+type EntityLinkedSearchFieldProps = {
+	languageOptions: Array<LanguageOption>,
+	label: string,
+	type: string | Array<string>,
+	empty?: boolean,
+	error?: boolean,
+	tooltipText?: string
+};
+
+/**
+ * Presentational component. This component renders a selection field which
+ * uses the user input to search for entities on the server. It allows the
+ * selection of one or more of the returned entities.
+ *
+ * @param {Object} props - The properties passed to the component.
+ * @param {boolean} props.error - Passed to the ValidationLabel within the
+ *        component to indicate a validation error.
+ * @param {boolean} props.empty - Passed to the ValidationLabel within the
+ *        component to indicate that the field is empty.
+ * @param {Array<LanguageOption>} props.languageOptions - The list of possible languages for an
+ *        entity.
+ * @param {string | Array<string>} props.type - Determines the types of entity
+ *        to search for.
+ * @param {string} props.label - The text to be used for the input label.
+ * @returns {Object} A React component containing the rendered input.
+ */
+function EntityLinkedSearchField(
```

Comment:
I would suggest naming this component to reflect its more general purpose, maybe something like EntitySearchFieldOption ?